artifactory/commands/generic/search.go

```diff
@@ -22,93 +26,125 @@ type SearchResult struct {
 
 type SearchCommand struct {
 	GenericCommand
-	searchResult []SearchResult
 }
 
 func NewSearchCommand() *SearchCommand {
 	return &SearchCommand{GenericCommand: *NewGenericCommand()}
 }
 
-func (sc *SearchCommand) SearchResult() []SearchResult {
-	return sc.searchResult
-}
-
-func (sc *SearchCommand) SearchResultNoDate() []SearchResult {
-	for i, _ := range sc.SearchResult() {
-		sc.searchResult[i].Created = ""
-		sc.searchResult[i].Modified = ""
-		delete(sc.searchResult[i].Props, "vcs.url")
-		delete(sc.searchResult[i].Props, "vcs.revision")
+func SearchResultNoDate(reader *content.ContentReader) (*content.ContentReader, error) {
+	writer, err := content.NewContentWriter("results", true, false)
+	if err != nil {
+		return nil, err
 	}
-	return sc.searchResult
+	defer writer.Close()
+	for resultItem := new(SearchResult); reader.NextRecord(resultItem) == nil; resultItem = new(SearchResult) {
+		if err != nil {
+			return nil, err
+		}
+		resultItem.Created = ""
+		resultItem.Modified = ""
+		delete(resultItem.Props, "vcs.url")
+		delete(resultItem.Props, "vcs.revision")
+		writer.Write(*resultItem)
+	}
+	if err := reader.GetError(); err != nil {
+		return nil, err
+	}
+	reader.Reset()
+	return content.NewContentReader(writer.GetFilePath(), writer.GetArrayKey()), nil
 }
 
 func (sc *SearchCommand) CommandName() string {
 	return "rt_search"
 }
 
 func (sc *SearchCommand) Run() error {
-	return sc.Search()
+	reader, err := sc.Search()
+	sc.Result().SetReader(reader)
+	return err
 }
 
-func (sc *SearchCommand) Search() error {
+func (sc *SearchCommand) Search() (*content.ContentReader, error) {
 	// Service Manager
 	rtDetails, err := sc.RtDetails()
 	if errorutils.CheckError(err) != nil {
-		return err
+		return nil, err
 	}
 	servicesManager, err := utils.CreateServiceManager(rtDetails, false)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	// Search Loop
 	log.Info("Searching artifacts...")
-	var resultItems []clientutils.ResultItem
+	var searchResults []*content.ContentReader
 	for i := 0; i < len(sc.Spec().Files); i++ {
-
 		searchParams, err := GetSearchParams(sc.Spec().Get(i))
 		if err != nil {
 			log.Error(err)
-			return err
+			return nil, err
 		}
 
-		currentResultItems, err := servicesManager.SearchFiles(searchParams)
+		currentReader, err := servicesManager.SearchFiles(searchParams)
 		if err != nil {
-			return err
+			log.Error(err)
+			return nil, err
 		}
-		resultItems = append(resultItems, currentResultItems...)
+		searchResults = append(searchResults, currentReader)
 	}
-
-	sc.searchResult = aqlResultToSearchResult(resultItems)
-	clientutils.LogSearchResults(len(resultItems))
-	return err
+	defer func() {
+		for _, reader := range searchResults {
```

Comment:
Similar to a previous comment I added here, it is best to add the defer right after *currentReader* is received, and not in a loop here.